### DISCUSSION TITLE
How to transfer the ownership of a move module?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#43] [Questions] [answered]
Asked on discord by `dellwatson`

since smart contract is on account, then is it not possible to transfer the smart contract? 
instead i should transfer the resources (struct) to other account? how this move smart contract logic works in case i want to move smart contract to other person

**Comment by 0x-j** [comment, marked as answer]
Similar to Solidity contract, although a module is published under an address (could be an EOA account or resource account or Object), you can program the admin of the module. [Checkout the ANS router module, where it only allows current admin to update admin](https://github.com/aptos-labs/aptos-names-contracts/blob/main/router/sources/router.move#L78-L97).
If you want to transfer ownership of a module to another person, you would update the module admin you stored to that person.

**Comment by MoonShiesty** [comment]
@0xaptosj nice example

> could be an EOA account or resource account or **Object**

i believe its not possible to publish a module as an Object currently, might be wrong about this

module objects will really nice - everyone won't re-implement the concept of an owner in each module

**Reply by 0x-j** [reply]
Yeah i think publish module under resource is still WIP, cc @JohnChangUK 

Today we usually publish module under our own account (i.e. EOA account)

**Reply by MoonShiesty** [reply]
looking forward to module objects, big QoL improvement

**Reply by JohnChangUK** [reply]
Yes coming very soon! This will allow modules to be contained in separate objects, and allowing upgrades to be done via ownership of the object/code.

**Reply by MoonShiesty** [reply]
can't wait! as soon as this ships, there's at least a hundred lines of duplicated code im deleting
